pennylane/tape/tape.py

```diff
@@ -1555,8 +1582,18 @@ def copy(self, copy_operations=False):
             tape._ops = self._ops.copy()
             tape._measurements = self._measurements.copy()
 
-        tape._update()
+        tape._graph = None
+        tape._specs = None
+        tape._depth = None
```

Comment:
Maybe instead of resetting these to `None`, we copy their values too:
```suggestion
        tape._graph = self._graph
        tape._specs = self._specs
        tape._depth = self._depth
```